### PR TITLE
[BACKLOG-8284] Convert filter classes to model classes

### PR DESCRIPTION
**Pull request by dcleao**
@pentaho/millenniumfalcon please review.

@jheck123: this is the only commit with docs: https://github.com/pentaho/pentaho-platform-plugin-common-ui/commit/f6568fb5dfec6933e44af36c3b6274590e43c7f9

### CODE REVIEW

**Comment by dcleao** [comment]
Ok devs, this is good to go when your review is finished!

File: package-res/resources/web/pentaho/type/filter/_core/tree.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+       * Gets the inverse filter class.
+       *
+       * @name _inverseClass
+       * @memberOf pentaho.type.filter.Tree#
+       * @type {Class.<pentaho.type.filter.Tree>}
+       * @protected
+       * @abstract
+       * @readOnly
+       */
+
```
Comment (carlosrusso) [inline package-res/resources/web/pentaho/type/filter/_core/tree.js]: blank line

File: package-res/resources/web/pentaho/type/filter/_core/tree.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            } else {
+              ops.push(op);
+            }
+          }
+        }
+
+        switch(ops.length) {
+          // Any non-null arguments?
+          case OP: return this;
```
Comment (carlosrusso) [inline package-res/resources/web/pentaho/type/filter/_core/tree.js]: I wasn't aware that assigning a variable to the `case` statement was possible.

File: package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This namespace contains types for expressing
+ * [intensional]{@link https://en.wikipedia.org/wiki/Intensional_definition}.
+ * filters of complex values or lists of.
```
Comment (carlosrusso) [inline package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc]: The phrasing "lists of" sounds a bit awkward to me. Did @jheck123 take a look at this?

File: package-res/resources/web/pentaho/type/filter/_core/tree.js (hunk 1 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/type/filter/_core/tree.js]: Actually, that's a virtual doclet and is not related to the following method, whose name is `negate`.

File: package-res/resources/web/pentaho/type/filter/_core/tree.js (hunk 2 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/type/filter/_core/tree.js]: Java, for example, only allows constants.

File: package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc (hunk 3 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc]: Just to make it clear, the intended meaning is "filters of complex values or of lists of complex values."

**Comment by carlosrusso** [comment]
👍  No more comments for now.

File: package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc (hunk 3 above)
Comment (jheck123) [inline package-res/resources/web/pentaho/type/filter/_doc/_namespace.jsdoc]: How about: filters of complex values or lists thereof.

File: package-res/resources/web/pentaho/data/Table.js [hunk 4]
```diff
@@ -15,357 +14,10 @@
...
 
-      return {model: attrSpecs, rows: rowSpecs};
-    }
-  });
+  "use strict";
 
-  return Table;
-});
+  return AbstractTable.core.Table;
+});
```
Comment (davidmsantos90) [inline package-res/resources/web/pentaho/data/Table.js]: Going with the momentum of the reviews in DET, there is a newline missing

**Comment by davidmsantos90** [comment]
Looking good :+1:

**Comment by dcleao** [comment]
Now good to press the button! No squashing needed.

**Pull request merged**